### DISCUSSION TITLE
[Feature] Family tree view, people relationships

### DISCUSSION THREAD

**Discussion by frejaya** [immich-app/immich#8044] [Feature Request]
### The feature

It would be amazing to be able to assign relationships between people and view them from a family tree perspective. One thing I find in really old family photos is that whoever cleaned up the photos each time often does it from their own perspective understandably (labeling a person as "great grandma"), when these relationships are obviously relative to whoever is viewing the photo. My great grandma will be different than my grandma's great grandma and so whoever labeled the photo knew who it was but it can be lost to time. I'd like to be able to set the relationships between people in the face recognition and then see them in a family tree. E.g.: spouse, child, parent, grandparent, great grandparent.

I've only seen this in one other photo tool and it looks so helpful! 

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by jonebarker** [comment]
This is a great idea.  I was thinking of an expandable grid similar to how the map page works, with a sidebar list of [searchable] family members that when selected, zooms into the individual.

**Comment by bo0tzz** [comment]
This is out of scope, Immich is not genealogy software.

**Comment by Coysh** [comment]
I think this is a great idea. I would use this. Slightly disagree with @bo0tzz - I don't think this is a request for a full genealogy implementation, just a basic relations for families

**Reply by frejaya** [reply]
Yes, what I was asking for is relationships between people, not genealogy. But I understand that it might be too far beyond what the Immich team is willing to build and they want to focus on what they do best and Immich's core offering. 

I was inspired by this photo organization software called LAMU, they have relationships. It's shown here and I just really wish I could do that in Immich. 

https://m.youtube.com/watch?v=U5eHTWPFN9E

**Reply by davidgascon** [reply]
100% agree with you too. It doesn't need to be extensive. I'm scanning in a ton of old photos from my great grandparents to archive, and I'd love to be able to show my kid photos of his great grandparents, or show my mom photos of her when she was a kid.

**Reply by ItsJamin** [reply]
I think even what is shown in the video is more than I would ask for. I dont need predefined relationships, are merging of persons to a couple. Just a named relationship between one person and another. That would be great!

**Comment by mertalev** [comment]
I think a concept of relationships could be neat, similar to setting a date of birth. It's another way of personalization and could be particularly nice for shared photos. A tree view specifically might be too much work, but just having and using that metadata in the UI can make for a more personal and interactive experience.

**Comment by bttfw** [comment]
Hope this will be picked up at some point would make immich better then Google 100 times

**Comment by Tricade** [comment]
I would feel that feature too

**Comment by ShailChoksi** [comment]
I went ahead and vibe coded something that works on top of Immich API:

https://github.com/ShailChoksi/treemich

Feel free to create PRs and ask for more features

**Comment by badbob001** [comment]
I would love this feature!

I think by adding hard relationships like (child of [one or two people]) + (married/??? to), the system can generate soft relationships to fill in the rest. Then it can have a family tree view where it'll default the focus on YOU and show the relative title of others (ie: what you call them and what they call you back in reverse). And you can change the person in focus so you can tell your kids how to address so and so.

I guess it can balloon to relationships like direct friends (which leads to friend of a friend), roommates, classmates, co-workers, something something kevin bacon, but maybe too soon for that.
